Add returned letters callback (#263)
alphagov/notifications-tech-docs: source/documentation/_callbacks.md

|#`destination_number` | The number the message was sent to (your number) | 07700987654|
|#`message` | The received message | Hello Notify!|
|#`date_received` | The UTC datetime that the message was received by Notify | 2017-05-14T12:15:30.000000Z|

### Returned letters

> When a letter you sent is returned, Notify will send details of the returned letter to your callback url.

Reviewer: ```suggestion
When a letter you sent is returned, Notify will send details of the returned letter to your callback URL.
```

Find more [information about returned letters](https://www.notifications.service.gov.uk/using-notify/delivery-times#returned-mail).  It can take a few weeks to receive information about a returned letter.

> The callback message is formatted in JSON. All of the values are strings. The key, description and format of the callback message arguments will be:

Reviewer: ```suggestion
The callback message is formatted in JSON. The key, description and format of the callback message arguments will be:
```

| Key | Description | Format |
|:---|:---|:---|

> | #`notification_id` | Notify’s id for the returned letter | UUID |

Reviewer: We capitalise this elsewhere in the docs
```suggestion
| #`notification_id` | Notify’s ID for the returned letter | UUID |
```

> | #`reference` | The reference sent by the service | `12345678` or null |

Reviewer: We don’t say _or null_ for other fields which are nullable (for example `sent_by`). Maybe we should say _optional_ for these fields instead? We should avoid `null` specifically because only exists in some languages, for example Python has `None`, Ruby has `nil`.
```suggestion
| #`reference` | The reference sent by the service | `12345678` |
```

Reviewer: For consistency with the documentation of other callbacks I’m going to keep this the same and add null to the other fields

| #`date_sent` | The time the letter was sent | `2017-05-14T12:15:30.000000Z` |

> | #`sent_by` | The email address of the service member who sent the letter | `hello@gov.uk`|

Reviewer: ```suggestion
| #`sent_by` | The email address of the service member who sent the letter | `hello@gov.uk` or null |
```

| #`template_name` | The name of the template that was used | `Template name` |
| #`template_id` | The id of the template that was used | UUID |
| #`template_version` | The version number of the template that was used | `1` |

> | #`spreadsheet_file_name`| The name of the uploaded spreadsheet | `contact_list.csv` |

Reviewer: ```suggestion
| #`spreadsheet_file_name`| The name of the uploaded spreadsheet | `contact_list.csv`  or null |
```

> | #`spreadsheet_row_number`| The row in the spreadsheet  | `1` |

Reviewer: This field will never be `1`. Should we also say that?
```suggestion
| #`spreadsheet_row_number`| The row in the spreadsheet  | `2` |
```

Reviewer: ```suggestion
| #`spreadsheet_row_number`| The row in the spreadsheet  | `2` or null |
```

> | #`upload_letter_file_name`| The name of the uploaded letter | `uploaded_letter.pdf` |

Reviewer: ```suggestion
| #`upload_letter_file_name`| The name of the uploaded letter | `uploaded_letter.pdf`  or null |
```
